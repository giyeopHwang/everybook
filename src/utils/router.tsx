import { createBrowserRouter } from 'react-router-dom';

import Front from '@/components/front/Front.component';
import DashboardLayout from '@/components/dashboard/dashboard-layout/DashboardLayout.compoent';

export const router = createBrowserRouter([
  {
    path: '/',
    element: <Front />,
  },
  {
    path: 'dashboard',
    element: <DashboardLayout />,
    children: [
      {
        path: '',
        element: <div>가계부</div>,
      },
      {
        path: 'report',
        element: <div>보고서</div>,
      },
      {
        path: 'budget',
        element: <div>예산 관리</div>,
      },
      {
        path: 'account',
        element: <div>분류 관리</div>,
      },
    ],
  },
]);
