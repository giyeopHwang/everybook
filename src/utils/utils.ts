export const displayAmount = (
  amount: number,
  showAmount: boolean = true,
  placeholder: string = '*********'
) => {
  return showAmount ? amount.toLocaleString() : placeholder;
};
