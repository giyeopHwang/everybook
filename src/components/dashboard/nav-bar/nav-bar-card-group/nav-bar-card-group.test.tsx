import { renderWithWrapper } from '@/test/test.utils';
import { mockNet, mockPlus, mockMinus } from '@/test/test.mock';
import NavBarCardGroup from './nav-bar-card-group.component';

describe('NavBarCardGroup', () => {
  it('renders NavBarCardGroup component', () => {
    const navBarCardGroup = renderWithWrapper(
      <NavBarCardGroup net={mockNet} plus={mockPlus} minus={mockMinus} />
    );

    expect(navBarCardGroup).toMatchSnapshot();
  });

  it('renders collapsed when specified', () => {
    const navBarCardGroup = renderWithWrapper(
      <NavBarCardGroup
        net={mockNet}
        plus={mockPlus}
        minus={mockMinus}
        expanded={false}
      />
    );

    expect(navBarCardGroup).toMatchSnapshot();
  });
});
