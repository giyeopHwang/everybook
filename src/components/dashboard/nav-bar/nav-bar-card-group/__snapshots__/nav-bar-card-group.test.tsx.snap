// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBarCardGroup > renders NavBarCardGroup component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-7ttcrf"
      >
        <div
          class="css-135bpbl"
        >
          <header
            class="css-kvjl0w"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 9h16.5m-16.5 6.75h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                NET
              </h4>
            </div>
            <span>
              100
            </span>
          </header>
          <section
            class="css-1r2jysi"
          />
        </div>
        <div
          class="css-135bpbl"
        >
          <header
            class="css-1r6e9a4"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 4.5v15m7.5-7.5h-15"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                PLUS
              </h4>
            </div>
            <span>
              300
            </span>
          </header>
          <section
            class="css-1r2jysi"
          >
            <div
              class="css-uf1ume"
            >
              <span>
                └ 
                PLUS_ITEM_1
              </span>
              <span>
                100
              </span>
            </div>
            <div
              class="css-uf1ume"
            >
              <span>
                └ 
                PLUS_ITEM_2
              </span>
              <span>
                200
              </span>
            </div>
          </section>
        </div>
        <div
          class="css-135bpbl"
        >
          <header
            class="css-1i1w77g"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                MINUS
              </h4>
            </div>
            <span>
              200
            </span>
          </header>
          <section
            class="css-1r2jysi"
          >
            <div
              class="css-uf1ume"
            >
              <span>
                └ 
                MINUS_ITEM_1
              </span>
              <span>
                50
              </span>
            </div>
            <div
              class="css-uf1ume"
            >
              <span>
                └ 
                MINUS_ITEM_2
              </span>
              <span>
                150
              </span>
            </div>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-7ttcrf"
    >
      <div
        class="css-135bpbl"
      >
        <header
          class="css-kvjl0w"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 9h16.5m-16.5 6.75h16.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              NET
            </h4>
          </div>
          <span>
            100
          </span>
        </header>
        <section
          class="css-1r2jysi"
        />
      </div>
      <div
        class="css-135bpbl"
      >
        <header
          class="css-1r6e9a4"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              PLUS
            </h4>
          </div>
          <span>
            300
          </span>
        </header>
        <section
          class="css-1r2jysi"
        >
          <div
            class="css-uf1ume"
          >
            <span>
              └ 
              PLUS_ITEM_1
            </span>
            <span>
              100
            </span>
          </div>
          <div
            class="css-uf1ume"
          >
            <span>
              └ 
              PLUS_ITEM_2
            </span>
            <span>
              200
            </span>
          </div>
        </section>
      </div>
      <div
        class="css-135bpbl"
      >
        <header
          class="css-1i1w77g"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              MINUS
            </h4>
          </div>
          <span>
            200
          </span>
        </header>
        <section
          class="css-1r2jysi"
        >
          <div
            class="css-uf1ume"
          >
            <span>
              └ 
              MINUS_ITEM_1
            </span>
            <span>
              50
            </span>
          </div>
          <div
            class="css-uf1ume"
          >
            <span>
              └ 
              MINUS_ITEM_2
            </span>
            <span>
              150
            </span>
          </div>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`NavBarCardGroup > renders collapsed when specified 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-7ttcrf"
      >
        <div
          class="css-135bpbl"
        >
          <header
            class="css-kvjl0w"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 9h16.5m-16.5 6.75h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                NET
              </h4>
            </div>
            <span>
              100
            </span>
          </header>
        </div>
        <div
          class="css-135bpbl"
        >
          <header
            class="css-1r6e9a4"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 4.5v15m7.5-7.5h-15"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                PLUS
              </h4>
            </div>
            <span>
              300
            </span>
          </header>
        </div>
        <div
          class="css-135bpbl"
        >
          <header
            class="css-1i1w77g"
          >
            <div
              class="css-vmk80s"
            >
              <svg
                aria-hidden="true"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                style="width: 0.9rem; height: 0.9rem;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <h4
                class="css-pqtvky"
              >
                MINUS
              </h4>
            </div>
            <span>
              200
            </span>
          </header>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-7ttcrf"
    >
      <div
        class="css-135bpbl"
      >
        <header
          class="css-kvjl0w"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.75 9h16.5m-16.5 6.75h16.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              NET
            </h4>
          </div>
          <span>
            100
          </span>
        </header>
      </div>
      <div
        class="css-135bpbl"
      >
        <header
          class="css-1r6e9a4"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5v15m7.5-7.5h-15"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              PLUS
            </h4>
          </div>
          <span>
            300
          </span>
        </header>
      </div>
      <div
        class="css-135bpbl"
      >
        <header
          class="css-1i1w77g"
        >
          <div
            class="css-vmk80s"
          >
            <svg
              aria-hidden="true"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              style="width: 0.9rem; height: 0.9rem;"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <h4
              class="css-pqtvky"
            >
              MINUS
            </h4>
          </div>
          <span>
            200
          </span>
        </header>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
